server/src/main/java/org/apache/cassandra/sidecar/handlers/OpenApiHandler.java

```diff
@@ -0,0 +1,164 @@
+     */
+    private boolean isYamlRequest(RoutingContext context)
+    {
+        String path = context.request().path();
+        if (path != null && path.endsWith(".yaml"))
+        {
+            return true;
+        }
+        
+        String acceptHeader = context.request().getHeader("Accept");
+        return acceptHeader != null && acceptHeader.contains("application/yaml");
+    }
+    
+    /**
+     * Loads the generated OpenAPI specification from resources or build output
+     */
+    private String loadGeneratedOpenApiSpec(boolean isYaml) throws IOException
+    {
+        // First try to load from classpath resources (for packaged deployments)
+        String content = loadFromClasspathResource(isYaml);
+        if (content != null)
+        {
+            return content;
+        }
+
+        // Fallback to file system paths (for development)
+        return loadFromFileSystem(isYaml);
+    }
+
+    /**
+     * Attempts to load the OpenAPI spec from classpath resources
+     */
+    private String loadFromClasspathResource(boolean isYaml) throws IOException
+    {
+        String fileName = isYaml ? "openapi.yaml" : "openapi.json";
+        String resourcePath = "/openapi/" + fileName;
+        
+        InputStream inputStream = getClass().getResourceAsStream(resourcePath);
+        if (inputStream == null)
+        {
+            return null;
+        }
+        
+        logger.debug("Loading OpenAPI specification from classpath: {}", resourcePath);
+        try 
+        {
+            byte[] bytes = inputStream.readAllBytes();
+            String content = new String(bytes, StandardCharsets.UTF_8);
+            logger.info("Loaded OpenAPI specification from classpath resource");
+            return content;
+        }
+        finally
+        {
+            inputStream.close();
+        }
+    }
+
+    /**
+     * Attempts to load the OpenAPI spec from file system (development mode)
+     */
+    private String loadFromFileSystem(boolean isYaml) throws IOException
+    {
+        String fileName = isYaml ? "openapi.yaml" : "openapi.json";
+        String[] possiblePaths = {
+            "server/build/generated/openapi/" + fileName,    // From project root
+            "build/generated/openapi/" + fileName,           // From server directory
+            "../server/build/generated/openapi/" + fileName, // Alternative relative path
+        };
+        
+        Path foundPath = null;
+        for (String pathStr : possiblePaths)
+        {
+            Path path = Paths.get(pathStr);
+            if (Files.exists(path))
+            {
+                foundPath = path;
+                break;
+            }
+        }
+        
+        if (foundPath == null)
+        {
+            throw new IOException("Generated OpenAPI specification not found in classpath resources or file system. " +
+                                "Please run './gradlew openApiGenerate' and ensure the generated file is included in the build.");
```

Comment:
Please update the gradle task name here as well. Or just mention that read the document on generating the openAPI specs w/o mention a specific gradle task name. Would defer to your choice.